util/XMLDoc.cpp

```diff
@@ -103,7 +103,7 @@ const XMLElement& XMLElement::Child(const std::string& tag) const {
         [&tag] (const XMLElement& e) { return e.m_tag == tag; });
 
     if (match == children.end())
-        throw NoSuchChild("XMLElement::Child(): The XMLElement \"" + Tag() + "\" contains no child \"" + tag + "\".");
+        throw std::out_of_range("XMLElement::Child(): The XMLElement \"" + Tag() + "\" contains no child \"" + tag + "\".");
```

Comment:
Is `out_of_range` appropriate in this context? The children aren't stored in any particular order, I think... Maybe `invalid_argument` would fit better? (I'm not sure... feel free to disregard...)

Reply:
I'm following the general semantics of `std::map::at` here.  `XMLElement::Child` returns an element accoring to a key, similar to `std::map::at`. Because `XMLElement::Child` has no other way to expose a non found child (e.g. end() iterator) due to the return type, `std::out_of_range` is thrown.